### PR TITLE
vault{,-bin}: 1.7.3 -> 1.8.0

### PR DESCRIPTION
**Pull request by amaxine**
###### Motivation for this change
https://github.com/hashicorp/vault/releases/tag/v1.8.0

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Chili-Man — APPROVED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 131874` at a4fe10f9 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built successfully:</summary>
  <ul>
    <li>vault</li>
    <li>vault-bin</li>
  </ul>
</details>

**Review by Chili-Man — APPROVED** [review]

File: pkgs/tools/security/vault/default.nix [hunk 1]
```diff
@@ -1,27 +1,35 @@
...
 
-  buildFlagsArray = [ "-tags=vault" "-ldflags=-s -w -X ${goPackagePath}/sdk/version.GitCommit=${src.rev}" ];
+  preBuild = ''
+    buildFlagsArray+=(
+      "-tags=vault"
+      "-ldflags=\
+        -s -w \
+        -X github.com/hashicorp/vault/sdk/version.GitCommit=${src.rev} \
+        -X github.com/hashicorp/vault/sdk/version.Version=${version} \
+        -X github.com/hashicorp/vault/sdk/version.VersionPrerelease=")
```
Comment (SuperSandro2000) [inline pkgs/tools/security/vault/default.nix:31]:
```suggestion
    buildFlagsArray+=(
      "-tags=vault"
    )

    ldflags = [
      "-s" "-w"
      "-X github.com/hashicorp/vault/sdk/version.GitCommit=${src.rev}"
      "-X github.com/hashicorp/vault/sdk/version.Version=${version}"
      "-X github.com/hashicorp/vault/sdk/version.VersionPrerelease="
```

Comment (maxeaubrey) [reply pkgs/tools/security/vault/default.nix:31]:
```
/nix/store/bmxafbqrlvcwpcwd21irmirid4rkpank-stdenv-linux/setup: line 91: ldflags: command not found
```

**Review by Chili-Man — APPROVED** [review]

**Review by bobrik — APPROVED** [review]

File: pkgs/tools/security/vault/default.nix (hunk 1 above)
Comment (zowoq) [reply pkgs/tools/security/vault/default.nix:31]: Need to move them out of `preBuild`.
```diff
diff --git a/pkgs/tools/security/vault/default.nix b/pkgs/tools/security/vault/default.nix
index d502cc779c2..e5b8b370ff3 100644
--- a/pkgs/tools/security/vault/default.nix
+++ b/pkgs/tools/security/vault/default.nix
@@ -21,15 +21,14 @@ buildGoModule rec {
 
   nativeBuildInputs = [ installShellFiles makeWrapper ];
 
-  preBuild = ''
-    buildFlagsArray+=(
-      "-tags=vault"
-      "-ldflags=\
-        -s -w \
-        -X github.com/hashicorp/vault/sdk/version.GitCommit=${src.rev} \
-        -X github.com/hashicorp/vault/sdk/version.Version=${version} \
-        -X github.com/hashicorp/vault/sdk/version.VersionPrerelease=")
-  '';
+  buildFlagsArray = [ "-tags=vault" ];
+
+  ldflags = [
+    "-s" "-w"
+    "-X github.com/hashicorp/vault/sdk/version.GitCommit=${src.rev}"
+    "-X github.com/hashicorp/vault/sdk/version.Version=${version}"
+    "-X github.com/hashicorp/vault/sdk/version.VersionPrerelease="
+  ];
 
   postInstall = ''
     echo "complete -C $out/bin/vault vault" > vault.bash
```

**Comment by risicle** [comment]
`nixpkgs-review` happy, macos 10.15, nixos x86_64, `nixosTests.vault` & `nixosTests.vault-postgresql` pass.

File: pkgs/tools/security/vault/default.nix (hunk 1 above)
Comment (maxeaubrey) [reply pkgs/tools/security/vault/default.nix:31]: Done, but considering both uses are frequent enough in `nixpkgs`... :confused:

**Pull request merged**